auditor/perturbations/base.py

```diff
@@ -0,0 +1,19 @@
+from abc import ABC, abstractmethod, abstractproperty
+from typing import List
+
+
+class AbstractPerturbation(ABC):
+    """Abstract class to aid in creation of perturbation classes
+    """
+    def __init__(self) -> None:
+        super().__init__()
+
+    @abstractmethod
```

Comment:
maybe add a comment explaining what derived concrete classes should do with this method ?